src/ir/proto/raksha_type.proto

```diff
@@ -0,0 +1,100 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+// This proto file contains the proto representation of the Raksha IR. This
+// allows frontends to construct the Raksha IR in any language and send the IR
+// over the wire or across an FFI layer for analysis.
+
+syntax = "proto3";
+
+package raksha.type.proto;
+
+message TypeProto {
```

Comment:
I'm not sure if we should even keep around this `TypeProto` file. While we are making some minimal use of encoding and decoding these protos in Raksha types, we aren't actually using types for anything in Raksha at the moment. It might be best to just have a clean break from the Arcs conception of types and build our own types as we need them. Otherwise, we risk sleepwalking into following the Arcs path, which might not always work for us.

For instance, does it make sense to think of a `BIGINT` as a primitive type? Or a `DURATION`? This is one of those things where the answer is "maybe, maybe not". In both the Souffle and potential custom solver case, the only primitive types (in this case meaning those that map directly to a simple type in the underlying runtime) would be `string`, `int64_t`, `uint64_t`, and `double`.

Reply:
I think it might be a good idea to simplify the proto to just the things we use currently.